app/Platform/Actions/MigrateAchievementIdsToDifferentGameId.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Actions;
+
+use App\Community\Enums\ArticleType;
+use App\Models\Achievement;
+use App\Models\Game;
+use App\Models\PlayerAchievement;
+use App\Models\PlayerGame;
+use App\Models\User;
+use App\Platform\Jobs\UpdateGameMetricsJob;
+use App\Platform\Jobs\UpdatePlayerGameMetricsJob;
+use Illuminate\Support\Str;
+
+class MigrateAchievementIdsToDifferentGameId
+{
+    public function execute(array $achievementIds, int $gameId, User $user): void
+    {
+        // Determine which game(s) the achievements are coming from.
+        $oldGameIds = Achievement::whereIn('ID', $achievementIds)->select(['GameID'])->distinct()->pluck('GameID');
+
+        // Associate the achievements to the new game.
+        Achievement::whereIn('ID', $achievementIds)->update(['GameID' => $gameId]);
+
+        // Add an audit comment to the new game.
+        addArticleComment(
+            'Server',
+            ArticleType::GameModification,
+            $gameId,
+            "$user migrated " . Str::plural('achievement', count($achievementIds)) . ' ' .
+                implode(',', $achievementIds) . ' from ' .
+                Str::plural('game', count($oldGameIds)) . ' ' . $oldGameIds->implode(',') . '.'
+        );
+
+        // Ensure player_game entries exist for the new game for all affected users.
+        foreach (PlayerAchievement::whereIn('achievement_id', $achievementIds)->select(['user_id'])->distinct()->pluck('user_id') as $userId) {
+            if (!PlayerGame::where('game_id', $gameId)->where('user_id', $userId)->exists()) {
+                $playerGame = new PlayerGame(['user_id' => $userId, 'game_id' => $gameId]);
+                $playerGame->save();
+                dispatch(new UpdatePlayerGameMetricsJob($userId, $gameId));
```

Comment:
The `PlayerGameMetrics` should be updated for the user whether or not they've previously played the game. The player metrics for the old game(s) should also be updated.

Reply:
Updated in latest:
```php
// Update all affected player game metrics.
$affectedUserIds = PlayerAchievement::whereIn('achievement_id', $achievementIds)
    ->select(['user_id'])
    ->distinct()
    ->pluck('user_id');
foreach ($affectedUserIds as $userId) {
    // Ensure player_game entries exist for the new game for all affected users.
    if (!PlayerGame::where('game_id', $gameId)->where('user_id', $userId)->exists()) {
        $playerGame = new PlayerGame(['user_id' => $userId, 'game_id' => $gameId]);
        $playerGame->save();
    }

    dispatch(new UpdatePlayerGameMetricsJob($userId, $gameId));
    foreach ($oldGameIds as $oldGameId) {
        dispatch(new UpdatePlayerGameMetricsJob($userId, $oldGameId));
    }
}
```